src/components/locations/components/NewLocation.js

```diff
@@ -1,79 +1,97 @@
-                console.log(file, fileList);
-            }
-        },
-        // defaultFileList: [...fileList],
+        name: 'locationImage',
     };
     const renderFormItem = () => FORM_ITEMS.map(item => (
-        <Form.Item key={item.formControlName} hasFeedback>
+        <Form.Item label={item.placeholder} key={item.formControlName} hasFeedback>
             {getFieldDecorator(item.formControlName, {
                 rules: [{ message: item.message, required: item.required }],
-            })(<Input
-                type={item.type}
-                placeholder={item.placeholder}
-            />)}
+            })(item.type === 'textarea' ? (
+                <Input.TextArea />)
+                : (<Input type={item.type} />))}
         </Form.Item>
     ));
+    const dummyRequest = ({ onSuccess }) => {
+        setTimeout(() => {
+            onSuccess('ok');
+        }, 100);
+    };
     return (
         <div className="card_background">
             <Row type="flex" justify="center" align="middle">
                 <Col xs={20} md={18} lg={10}>
                     <Card
-                        className="text-center"
                         title={CREATE_NEW_LOCATION}
                     >
-                        <Form onSubmit={handleSubmit}>
+                        <Form {...FORM_ITEM_LAYOUT} onSubmit={handleSubmit}>
                             {renderFormItem()}
-                            <Form.Item>
-                                <Upload multiple accept="image/*" {...props}>
-                                    <Button>
-                                        <Icon type="upload" />
-                                        {UPLOAD_TEXT}
-                                    </Button>
-                                </Upload>
+                            <Form.Item label="Upload">
+                                {getFieldDecorator('images', {
+                                    getValueFromEvent: normFile,
+                                    rules: [{
+                                        message: 'Atleast one image should be uploaded',
+                                        required: true,
+                                    }],
+                                    valuePropName: 'fileList',
+                                })(
+                                    <Upload
+                                        supportServerRender
+                                        multiple
+                                        accept="image/*"
+                                        customRequest={dummyRequest}
+                                        {...props}
+                                    >
+                                        <Button>
+                                            <Icon type="upload" />
+                                            {UPLOAD_TEXT}
+                                        </Button>
+                                    </Upload>
```

Comment:
i don't understand what you are doing here. Why are you wrapping a `Button ` around a `Upload`?

Reply:
Used to trigger the upload section popup